images/devtools-terraform-v1beta1/context/Makefile

```diff
@@ -28,7 +28,7 @@ ifneq ($(tools_dir),)
 export PATH:=$(tools_dir):$(PATH)
 endif
 XDG_CACHE_HOME?=$(HOME)/.cache
-
+with_tfswitch=0
```

Comment:
I would rather go with:

```suggestion
WITH_TFSWITCH?=false
```

Reasoning for `true|false`:
- `[true](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/true.html)|[false](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/false.html)` is part of posix as commands so this variable can then be used in shell to determine the result.
- This is also valid json so can be used with json aware tooling.
- I feel it is inherently less ambiguous than using an integer, as when using integers questions arise as to what should happen if it is `3` for example.
- often with environment variables a blank variable is used to indicate false/unset and a set variable is used to indicate true, this further makes it unclear what to expect with =0.

Reasoning for uppercase:
- https://www.gnu.org/software/make/manual/make.html#Using-Variables
 > It is traditional to use upper case letters in variable names, but we recommend using lower case letters for variable names that serve internal purposes in the makefile, and reserving upper case for parameters that control implicit rules or for parameters that the user should override with command options (see [Overriding Variables](https://www.gnu.org/software/make/manual/make.html#Overriding)).

Reasoning for `?=`:
 - If you use `WITH_TFSWITCH=false` - then it cannot be overwritten by environment variable, as the value you set here will take precedence. `?=` only sets it if not yet set, so it then will take the value from the environment if it is set [[ref](https://www.gnu.org/software/make/manual/make.html#Flavors)]:
  > There is another assignment operator for variables, ‘?=’. This is called a conditional variable assignment operator, because it only has an effect if the variable is not yet defined. This statement:

Reply:
I should also note, making it possible to use environment variables makes it easier to override using docker features and makes it less coupled to make.